### PR TITLE
new features and fixes

### PR DESCRIPTION
**Pull request by pikapika2020**
- `-parsejson` command-line option.
    Should be used in conjunction with `-json` (it works only this way). Possibility of JSON parsing if logs are in JSON format (field is called "log"; if a log string can not be parsed, then it prints string as usual in "msg" field)
- Now program is less verbose (should use `-debug` for verbosity)
- Data race fix (overlapping was envolved)
- A few more tests added and/or modified (BATS included)
- go 1.14 - modules now - `go build` does all the job, therefore...
- ...Makefile changed
- Now it works with latest cronexpr (despite it's old as hell anyways)
- Minor crontab.go fix (it was a problem with tests involved)
- Minor main.go stylistic edits

(more fixes ahead in next PRs, hopefully I will have time for that)

### CODE REVIEW

**Comment by pikapika2020** [comment]
Should edit .travis.yml also...

**Comment by krallin** [comment]
Hi there, and thanks for taking the time to contribute.

Unfortunately, it's difficult to reason about the changes you're proposing here. You mention having fixed some bugs, but you're not describing what the bugs were. Your PR also mixes a number of different changes, including new functionality, bug fixes, build process tweaks, and non-functional stylistic changes.

Do you mind taking the time to:

- Split this up in individual commits for each change you'd like to propose? It's possible that we'd like to merge some of your changes in, but not others (notably: bug fixes make sense to merge, but stylistic changes perhaps not, and new functionality might need some discussion)
- For the bugs you fixed, take the time to describe what the bug was, and why your fix works? As-is, I would otherwise have to try and guess at what your intentions were, which is a bit of a waste of time for everyone involved.

Thanks!

**Comment by pikapika2020** [comment]
> Unfortunately, it's difficult to reason about the changes you're proposing here.

In my opinion, this PR is not too large to take a long time to review. I want to save some time while contributing too. But alright, got your point here. Going to introduce changes gradually one by one in the next couple of weeks, hopefully.

But I have an important question for you:
_Will you accept transition to `go 1.14` and official go module system (which is stable since version `1.13`) with the latest updates of all project dependicies and corresponding changes to project source code (as I already did in this PR, all tests are okay). Or is it another one thing for discussion?_

Or are you going to accept just bugfixes and new features, so there is no point for me to introduce some changes to build process with the new version of `go`?

I would like to commit like so (also that's a short description for the current PR):
1. Transition to `go 1.14` and go official module system (`go mod`) with all corresponding fixes to code and tests
2. Data race fix
3. Feature flag `-parsejson` which allow users to have parsed JSON data in program output
4. Verbosity fix - now program output does not contain things like `...job is still running since...` and `...job took too long to run: it should have started...` without `-debug` flag

No. 4 is still discussable, I guess, because probably some got used to this verbosity (still discussable - as issue #57 shows). It can be implemented as feature flag, ofc.

Also you always can release new version from any branch as `...-beta`, allowing users to give their feedback, and after that all flag/fields renames/etc. can take place

**Comment by krallin** [comment]
@pikapika2020,  we'll definitely incorporate build modernization that updates the codebase to new Go modules, as well as bug fixes. 

I think 3 is fine too, though it'd be preferable if we could avoid using globals to implement it. We can discuss it on a PR for it though.

For 4, I do disagree and I'd rather not that merge that in. Those messages are warnings that are actually pretty important for new users. We can have a quiet mode that doesn't show them, but I think it's reasonable for them to show by default.

**Comment by pikapika2020** [comment]
@krallin, I thought exactly the same way actually!
+1 to avoid globals (going to discuss at some point), but it will refactoring, possibly a big one.
I also wanted to implement quiet (or "silent" mode), but it can entail big refactoring as well

**Comment by pikapika2020** [comment]
Closed in favor of #59

File: cron/cron.go [hunk 1]
```diff
@@ -63,6 +74,19 @@ func startReaderDrain(wg *sync.WaitGroup, readerLogger *logrus.Entry, reader io.
...
+	err := json.Unmarshal(line, &someJson)
+	if err != nil {
+		readerLogger.Info(string(line))
+	} else {
+		readerLogger.WithFields(logrus.Fields{
+			"log": someJson,
+		}).Info("json log data")
+	}
+}
+
```
Comment (yayayahei) [inline cron/cron.go:89]: This code helps me in my situation.

**Pull request closed without merging**